src/OmniSharp.Host/Endpoint/EndpointHandler.cs

```diff
@@ -5,31 +5,26 @@
 using System.Linq;
 using System.Reflection;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
 using Microsoft.CodeAnalysis;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using OmniSharp.Endpoint.Exports;
 using OmniSharp.Mef;
-using OmniSharp.Middleware.Endpoint.Exports;
 using OmniSharp.Models;
 using OmniSharp.Models.UpdateBuffer;
 using OmniSharp.Plugins;
+using OmniSharp.Stdio.Protocol;
 
-namespace OmniSharp.Middleware.Endpoint
+namespace OmniSharp.Endpoint
 {
-    class LanguageModel
+    public abstract class EndpointHandler
```

Comment:
Since you're making a public API, you might consider renaming from `Endpoint` to `EndPoint`.

Reply:
where did you get the extra capital `P` from?  I know I'm Canadian and everything... but I've never seen that! https://en.wikipedia.org/wiki/Endpoint

Reply:
I'm don't feel strongly about it. I just noticed that it's different than what we do in System.Net: https://msdn.microsoft.com/en-us/library/system.net.endpoint(v=vs.110).aspx

Reply:
😱